### ISSUE TITLE
Fail to fetch image in AWS S3 - wrong request header

### ISSUE THREAD

**Issue by BernardA** [vercel/next.js#23755]
### What version of Next.js are you using?

10.0.6

### What version of Node.js are you using?

14.15.4

### What browser are you using?

Chrome

### What operating system are you using?

mac Os

### How are you deploying your application?

in development

### Describe the Bug

Getting Access denied when trying to fetch images from S3, though S3 access policy seems correct.
Request is sent with headers below which do not seem correct given that it certainly is not same origin and it should in fact be CORS.
 Sec-Fetch-Site: same-origin
 Sec-Fetch-Mode: no-cors

### Expected Behavior

Should fetch images correctly. It actually does so if I remove any conditions from S3 access policy and leave it completely public, which could be expected.

### To Reproduce

see below

      <Image
            src={`${process.env.NEXT_PUBLIC_AWS_S3_ITEM_IMAGE}/${item.image.filename}`}
            alt={item.name}
            width="140"
            height="140"
        />

When loading the page with component above the following request is triggered: 

As you can see the headers mentioned above are included, though they do not seem pertinent.

<img width="529" alt="Screenshot 2021-04-06 at 18 04 45" src="https://user-images.githubusercontent.com/9449665/113742225-ae019480-9702-11eb-9c12-f1b7f01f60ad.png">

For information, the S3 bucket policy is as below: 

     {
    "Version": "2012-10-17",
    "Id": "http referer allow websites",
    "Statement": [
        {
            "Sid": "Allow get requests originating from localhost:3000 and quiamo.vote.",
            "Effect": "Allow",
            "Principal": "*",
            "Action": [
                "s3:GetObject",
                "s3:GetObjectVersion"
            ],
            "Resource": "arn:aws:s3:::quiamo-item-images/*",
            "Condition": {
                "StringLike": {
                    "aws:Referer": [
                        "http://localhost:3000*",
                        "http://127.0.0.1:3000*",
                        "https://quiamo.vote*",
                        "127.0.0.1:3000*"
                    ]
                }
            }
        }
    ]
    }

**Comment by mercteil** [comment]
```javascript
  images: {
    deviceSizes: [320, 544, 768, 1024, 1600, 1900, 1024],
    domains: ['mercteildev.s3.eu-central-1.amazonaws.com', 'mercteil.com'],
  },

**Comment by BernardA** [comment]
Yup! As said it fetches the images if I leave the S3 bucket as public with no conditions:

Here is my next.config on images

      images: {
        domains: [
            'mywebsite.com',
            'localhost',
            'mybucket1.s3-sa-east-1.amazonaws.com',
            'mybucket2.s3-sa-east-1.amazonaws.com',
            'mybucket3.s3-sa-east-1.amazonaws.com',
        ],
    },

**Comment by styfle** [comment]
Are you expecting the Image Optimization API to set a `referer` header?

Currently it doesn't sent any headers, it just requests the absolute url like so:

https://github.com/vercel/next.js/blob/9fcfce3463095af63e93251ae12f5cac8344aa7f/packages/next/next-server/server/image-optimizer.ts#L183

**Comment by Innei** [comment]
Is there a solution here?
